lib/public/components/common/form/inputs/DateTimeInputModel.js

```diff
@@ -0,0 +1,316 @@
+            && (this._max === null || this._value <= this._max);
+    }
+
+    /**
+     * Returns the raw input values
+     *
+     * @return {DateTimeInputRawData} the raw values
+     */
+    get raw() {
+        return this._raw;
+    }
+
+    /**
+     * Returns the current date represented by the inputs (null if no valid value is represented)
+     *
+     * @return {number|undefined} the current value
+     */
+    get value() {
+        return this._value;
+    }
+
+    /**
+     * Set the current date represented by the inputs (do not update the raw inputs)
+     *
+     * @param {number|undefined} value the new value
+     */
+    set value(value) {
+        if (this._value === value) {
+            return;
+        }
+
+        this._value = value;
+    }
+
+    /**
+     * Set the current date represented by the inputs and update the raw values accordingly
+     *
+     * @param {number|undefined} value the new value
+     * @param {boolean} [silent] if true, observers will not be notified of the change
+     * @return {void}
+     */
+    setValue(value, silent) {
+        if (value === this._value) {
+            return;
+        }
+
+        this._value = value;
+        const newRaw = value !== undefined
+            ? this.formatDateForHTMLInput(value)
+            : { date: '', time: '' };
+
+        if (this._raw.date !== newRaw.date || this._raw.time !== newRaw.time) {
+            this._raw = newRaw;
+            !silent && this.visualChange$.notify();
+        }
+
+        !silent && this.notify();
+    }
+
+    /**
+     * Returns an observable notified any time a visual change occurred
+     *
+     * @return {Observable} the observable
+     */
+    get visualChange$() {
+        return this._visualChange$;
+    }
+
+    /**
+     * States if the input is required
+     *
+     * @return {boolean} true if the input is required
+     */
+    get isRequired() {
+        return this._isRequired;
+    }
+
+    /**
+     * The minimum timestamp (in ms) allowed to be represented by the input
+     *
+     * @return {number|null} the minimum value
+     */
+    get min() {
+        return this._min;
+    }
+
+    /**
+     * Defines the minimum timestamp allowed to be represented by the input
+     *
+     * @param {number|null} value the minimum timestamp (in ms)
+     */
+    set min(value) {
+        this._min = value;
+    }
+
+    /**
+     * The maximum timestamp (in ms) allowed to be represented by the input
+     *
+     * @return {number|null} the maximum value
+     */
+    get max() {
+        return this._max;
+    }
+
+    /**
+     * Defines the maximum timestamp allowed to be represented by the input
+     *
+     * @param {number|null} value the maximum timestamp (in ms)
+     */
+    set max(value) {
+        this._max = value;
+    }
+
+    /**
+     * Returns the min values to apply to the inputs
+     *
+     * @return {(Partial<{date: string, time: string}>|string)} the min values
+     */
```

Comment:
I think it cannot return string and is it intentional not to use DateTimeInputRawData ?

Reply:
You're right about the string.
Yes it is intentional, because yes it happens to have the same properties but it's not the same semantic, it's not inputs raw data

Reply:
so as not the same maybe another typedef, because they looks quite similar at the first glance, at least for me it was a bit misleading

Reply:
TBH it's used only here, I think it would be overkill to create a typedef for this. If it need to be used anywhere else, I would extract a typedef.

About the typedef: they describe the structure of something specific, not just a structure. You should never reuse the structure just because it maches your type.

Reply:
ok